### ISSUE TITLE
3/30 내용

### ISSUE THREAD

**Issue by c2alpha** [crotielwf/SSUDL#1]
# Lab 3 Minimizing Cost
import tensorflow as tf
#tf.set_random_seed(777)  # for reproducibility

# tf Graph Input
X = [1, 2, 3]
Y = [1, 2, 3]

# Set wrong model weights
W = tf.Variable(1.0)

# Linear model
hypothesis = X * W

# cost/loss function
cost = tf.reduce_mean(tf.square(hypothesis - Y))  ---(A)

#sum이면 완전 다른 결과

#lossFunction이 CostFunction 포함

# Minimize: Gradient Descent Magic
optimizer = tf.train.GradientDescentOptimizer(learning_rate=0.1)
train = optimizer.minimize(cost)

# Launch the graph in a session.
sess = tf.Session()
# Initializes global variables in the graph.
sess.run(tf.global_variables_initializer())

for step in range(200):
    print(step, sess.run(W))
    sess.run(train)

===================================================================
(A) 에 cost 함수 보면 mean 으로 했을 경우와 sum으로 했을 경우에 차이가 있다.
sum을 사용할 경우 값의 변동성이 커서 learning rate를 더 잘 (작게) 잡아줘야 한다. 즉 크게(0.1)로 잡으면 w 값이 발산한다.
반대로 mean을 이용할 경우 초기 w를 크게 잡아도 최적화 된 값을 빠르게 찾아간다.
